ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/OptimizerController.java

```diff
@@ -207,5 +211,112 @@ public void scaleOutOptimizer(Context ctx) {
     optimizerManager.createResource(resource);
     ctx.json(OkResponse.of("success to scaleOut optimizer"));
   }
+
+  /**
+   * get optimizeGroups
+   * url = /optimize/resourceGroups/get
+   */
+  public void getResourceGroup(Context ctx) {
+    List<OptimizerResourceInfo> result =
+        optimizerManager.listResourceGroups().stream().map(group -> {
+          List<OptimizerInstance> optimizers = optimizerManager.listOptimizers(group.getName());
+          OptimizerResourceInfo optimizerResourceInfo = new OptimizerResourceInfo();
+          optimizerResourceInfo.setResourceGroup(optimizerManager.getResourceGroup(group.getName()));
+          optimizers.forEach(optimizer -> {
+            optimizerResourceInfo.addOccupationCore(optimizer.getThreadCount());
+            optimizerResourceInfo.addOccupationMemory(optimizer.getMemoryMb());
+          });
+          return optimizerResourceInfo;
+        }).collect(Collectors.toList());
+    ctx.json(OkResponse.of(result));
+  }
+
+  /**
+   * create optimizeGroup: name, container, schedulePolicy, properties
+   * url = /optimize/resourceGroups/create
+   */
+  public void createResourceGroup(Context ctx) {
+    Map<String, Object> map = ctx.bodyAsClass(Map.class);
+    String name = (String) map.get("name");
+    String container = (String) map.get("container");
+    Map<String, String> properties = (Map) map.get("properties");
+    ResourceGroup.Builder builder = new ResourceGroup.Builder(name, container);
+    builder.addProperties(properties);
+    optimizerManager.createResourceGroup(builder.build());
```

Comment:
We need to translate the primary key conflict exception error message into user-friendly prompts, such as "Group already exists."